static/js/plotjs/foldphase.js

```diff
@@ -18,6 +50,76 @@ for (let i = 0; i < n_labels; i++) {
     foldb.data.mjd_foldb[m] = folda.data.mjd_folda[m] + 1;
     foldberr.data.xs[m] = [foldb.data.mjd_foldb[m], foldb.data.mjd_foldb[m]];
   }
+
+  if (binsize > 1) {
+    var j;
+    var k;
+
+    var alist = [];
+    for (j = 0; j < folda.data.mjd_folda.length; j++) {
+      alist.push({
+        index: j,
+        mjd_folda: folda.data.mjd_folda[j],
+        mag: folda.data.mag[j],
+      });
+    }
+    alist.sort(function (a, b) {
+      return a.mjd_folda < b.mjd_folda
+        ? -1
+        : a.mjd_folda == b.mjd_folda
+        ? 0
+        : 1;
+    });
+    const amag_sort = [];
+    for (j = 0; j < alist.length; j++) {
+      amag_sort[j] = alist[j].mag;
+    }
+    const amag_sort_smooth = smoothing_func(amag_sort, binsize);
+    const amag_sort_smooth_reordered = [];
+    for (var j = 0; j < folda.data.mjd_folda.length; j++) {
+      for (var k = 0; k < folda.data.mjd_folda.length; k++) {
```

Comment:
There's no way to sort a list of tuples in JS? this double loop can't be good for you...

Reply:
You ask me questions like I know JS ;)